$NetBSD: patch-src_runtime_GNUmakefile,v 1.1 2018/10/17 12:59:49 leot Exp $

Add support to invoke paxctl(8) or similar programs to adjust PaX
permissions of src/runtime/sbcl during the build phase.

--- src/runtime/GNUmakefile.orig	2017-12-29 09:55:08.000000000 +0000
+++ src/runtime/GNUmakefile
@@ -14,6 +14,9 @@
 all: targets tags
 TARGET=sbcl
 
+# paxctl(8) or similar programs to adjust PaX permissions of src/runtime/sbcl
+SBCL_PAXCTL ?= :
+
 # Defaults which might be overridden or modified by values in the
 # Config file. Most of them are same on most systems right now.
 # If you need to override one of these, do it in Config.
@@ -78,6 +81,7 @@ targets: $(TARGET) $(OBJTARGET) sbcl.nm 
 
 $(TARGET): $(LIBSBCL)
 	$(CC) ${LINKFLAGS} -o $@ $(USE_LIBSBCL) $(LIBS)
+	$(SBCL_PAXCTL) $@
 
 # ld -r -o sbcl.o works on Linux, but not on other platforms.
 # On macOS, it fails to keep debug sections.
